src/asm/macro.c

```diff
@@ -98,6 +98,35 @@ char const *macro_GetArg(uint32_t i)
 					      : macroArgs->args[realIndex];
 }
 
+char const *macro_GetAllArgs(void)
+{
+	if (!macroArgs)
+		return NULL;
+
+	/*
+	 * Returning an empty string for no arguments
+	 * does not cooperate with beginExpansion,
+	 * so this returns a space.
+	 */
+	if (macroArgs->shift >= macroArgs->nbArgs)
+		return strdup(" ");
+
+	size_t len = strlen(macroArgs->args[macroArgs->shift]);
+
+	for (uint32_t i = macroArgs->shift + 1; i < macroArgs->nbArgs; i++)
+		len += 1 + strlen(macroArgs->args[i]);
+
+	char *str = malloc(len + 1);
+
+	strcpy(str, macroArgs->args[macroArgs->shift]);
```

Comment:
Arguably, no need to copy a terminator there

Reply:
The terminator is necessary so the `strcat`s work.

Reply:
`strcat` re-reads the input string to find its terminator, when we already reached the end of the previous string (which we copied). I'd prefer maintaining a destination pointer in a variable, and incrementing it always.

Reply:
Done. It still reads each macro arg three times: once to get its length for calculating the total `\#` length; once to get its length for appending with `memcpy`; and once to actually do the `memcpy`. A custom loop to copy one character at a time, incrementing the `ptr` each time, would avoid that third read, but I expect `strlen`+`memcpy` is faster regardless.

Reply:
That would need to be benchmarked, but I'll accept this for the time being. We can always come back to performance tuning later.